Tournament_Calculator.py

```diff
@@ -0,0 +1,323 @@
+#import modules
+import openpyxl
+import random
+import re
+
+#Player Name/Hero dictionary and set variable for below function
+y=0
+Player_List = {}
+
+
+#Function to populate dictionary with player name/Hero
+def Player_Creation(Hero):
+    global y
+    for row in sheet_obj.iter_rows(1):
+        for cell in row:
+            if cell.value == Hero:
+                z = (sheet_obj.cell(row=(cell.row), column=(2)).value)
+                x = 0
+                if z == 0:
+                    pass
+                else:
+                    while x < int(z):
+                        New_Player = "Player" + str(y)
+                        Player_List[New_Player] = Hero
+                        y +=1
+                        x +=1
+                return y
+
+ 
+
+#Set Excel defaults for player input
+path = "Input Data.xlsx"
+wb_obj = openpyxl.load_workbook(path, data_only=True)
+wb_obj.active = wb_obj['Data Input']
+sheet_obj = wb_obj.active
+
+#Populate Dictionary with Player names/Heroes
+Player_Creation("Arakni")
+Player_Creation("Azalea")
+Player_Creation("Betsy")
+Player_Creation("Bravo")
+Player_Creation("DIO")
+Player_Creation("Dash") 
+Player_Creation("Dorinthea")
+Player_Creation("Enigma")
+Player_Creation("Fai")
+Player_Creation("Kano") 
+Player_Creation("Kassai")
+Player_Creation("Katsu")
+Player_Creation("Kayo")
+Player_Creation("Levia")
+Player_Creation("Maxx")
+Player_Creation("Nuu")
+Player_Creation("Olympia")
+Player_Creation("Prism")
+Player_Creation("Rhinar") 
+Player_Creation("Riptide")
+Player_Creation("Boltyn")
+Player_Creation("Teklovossen")
+Player_Creation("Uzuri")
+Player_Creation("Victor") 
+Player_Creation("Viserai")
+Player_Creation("Vynnset")
+Player_Creation("Zen")
+
+#Set player total and rounds
+Month = ((sheet_obj.cell(row=(30),column=(2))).value)
+Players = ((sheet_obj.cell(row=(31),column=(2))).value)
+Sheet_Name = ((sheet_obj.cell(row=(36),column=(2))).value)
+Day2_Cut = ((sheet_obj.cell(row=(33),column=(2))).value)
+if Day2_Cut == str(True):
+    Day1 = ((sheet_obj.cell(row=(34),column=(2))).value)
+    Day2 = ((sheet_obj.cell(row=(35),column=(2))).value)
+else:
+    Rounds = sheet_obj.cell(row=(32),column=(2)).value
+
+#List for just player names/Heroes
+players = Player_List.fromkeys(Player_List)
+
+
+#Set defaults for Excel to Matchup Data
+path = "Talishar Meta Data.xlsx"
+wb_obj = openpyxl.load_workbook(path, data_only=True)
+wb_obj.active = wb_obj[Month]
+sheet_obj = wb_obj.active
+
+#dictionaries for winrates
+Matchups = {

```

Comment:
Data like this should never be stored in code, it eats up a ton of space and it's difficult to maintain. Definitely work on migrating this data into a data file of some sort, probably csv again since that can be read directly into pandas.

Long-term, you would want to have a script that can automatically pull this data from talishar instead of having to do it manually.